src/main/java/PieceFactory.java

```diff
@@ -38,8 +39,27 @@ public Piece create(String type) {
                 // King can never be captured so whatever
                 piece.setPointValue(0);
                 break;
+            default:
+                throw new IllegalArgumentException("Invalid piece type: " + type);
```

Comment:
As discussed a few days ago, we should further discuss Exceptions as a team (time permitting).